lib/ros_value.h

```diff
@@ -320,6 +323,29 @@ class RosValue {
     return values;
   }
 
+  // Used for python bindings
+  pybind11::buffer_info getPrimitiveArrayBufferInfo() {
+    if (getType() != Embag::RosValue::Type::array) {
+      throw std::runtime_error("Only arrays can be represented as buffers!");
+    }
+
+    const Embag::RosValue::Type type_of_elements = at(0)->getType();
```

Comment:
What is the value behind using our custom object here? There's no better type checking we can do?

Reply:
Might need to take this offline as I'm not sure what you mean here - the only way we can determine the type of the underlying data is checking the type of the RosValue, as that's where the definition is stored. All the typing is dynamic since we don't know the message types that we're going to be handling at compile time.